lib/chef/provider/package/snap.rb

```diff
@@ -62,27 +61,24 @@ def get_current_versions
           end
         end
 
-        def install_package(names, versions)
+        def install_package(names, _versions)
           if new_resource.source
             install_snap_from_source(names, new_resource.source)
           else
-            resolved_names = names.each_with_index.map { |name, i| available_version(i).to_s unless name.nil? }
-            install_snaps(resolved_names)
+            names.each { |pkg| install_snaps(pkg) }
```

Comment:
that does not look like the right way to fix this.

if the version is passed we should definitely be using that, so ignoring the argument is guaranteed to be buggy.

the problem with this looks like it is copying code from the dnf provider, where the available_version function does not return a version, but a name+version+arch tuple object called a Version, which when you call `#to_s` on it produces something like "name-version.arch" which can be passed to `dnf install name-version.arch`.

Version and available_version was probably the wrong word to use there.

Reply:
After did require analysis, I found, If we want to install a specific version of the snap package, then we can install a specific version with `channel` property like this: `sudo snap install kubectl --channel=1.6/stable --classic` , where `1.6` is the version and `stable` is the channel name.

so here we have two approaches:
1.) we need to pass `channel` property value with the version and there is no need to add version property. 
suppose if we want to install `node` package with `8` version, then we need to set `channal` value like this

```
snap_package 'install package' do
  package_name      ["node"]
  options           ["classic"]
  channel           "8/stable"
  action :install
end
```
OR

2.) Add new `version` property in the snap resource
In this approach, if we want to install `node` package with `8` version, then we need to set `version` value like this:
```
snap_package 'install package' do
  package_name      ["node"]
  options           ["classic"]
  version           "8"
  action :install
end
```
and also we need to update some of the code.